src/dwn/models/record.js

```diff
@@ -0,0 +1,244 @@
+
+  constructor(dwn, options = { }) {
+    this.#dwn = dwn;
+
+    // RecordsWriteMessage properties.
+    const { author, contextId = undefined, descriptor, recordId = null, target } = options;
+    this.#contextId = contextId;
+    if (descriptor?.data) delete descriptor.data;
+    this.#descriptor = descriptor ?? { };
+    this.#recordId = recordId;
+    
+    // Store the target and author DIDs that were used to create the message to use for subsequent reads, etc.
+    this.#author = author;
+    this.#target = target;
+    
+    // If the record `dataSize is less than the DwnConstant.maxDataSizeAllowedToBeEncoded value,
+    // then an `encodedData` property will be present.
+    this.#encodedData = options?.encodedData ?? null;
+
+    // If the record was created from a RecordsRead reply then it will have a `data` property.
+    if (options?.data) {
+      // this.#readableStream = isReadableWebStream(options.data) ? toIsomorphicReadableStream(options.data) : options.data;
+      this.#readableStream = isReadableWebStream(options.data) ? new ReadableWebToNodeStream(options.data) : options.data;
+    }
+  }
+
+  // Mutable Web5 Record Class properties.
+  get author() { return this.#author; }
+  get isFrozen() { return this.#isFrozen; }
+  get target() { return this.#target; }
+  set author(author) { this.#author = author; }
+  set target(target) { this.#target = target; }
+  
+  // Immutable DWN Record properties.
+  get id() { return this.#recordId; }
+  get contextId() { return this.#contextId; }
+  get dataFormat() { return this.#descriptor?.dataFormat; }
+  get dateCreated() { return this.#descriptor?.dateCreated; }
+  get interface() { return this.#descriptor?.interface; }
+  get method() { return this.#descriptor?.method; }
+  get parentId() { return this.#descriptor?.parentId; }
+  get protocol() { return this.#descriptor?.protocol; }
+  get recipient() { return this.#descriptor?.recipient; }
+  get schema() { return this.#descriptor?.schema; }
+
+  // Mutable DWN Record properties.
+  get dataCid() { return this.#descriptor?.dataCid; }
+  get dataSize() { return this.#descriptor?.dataSize; }
+  get dateModified() { return this.#descriptor?.dateModified; }
+  get datePublished() { return this.#descriptor?.datePublished; }
+  get published() { return this.#descriptor?.published; }
+
+  /**
+   * Data handling.
+   */
+
+  get data() {
+    if (!this.#encodedData && !this.#readableStream) {
+      // `encodedData` will be set if `dataSize` <= DwnConstant.maxDataSizeAllowedToBeEncoded. (10KB as of April 2023)
+      // `readableStream` will be set if Record was instantiated from a RecordsRead reply.
+      // If neither of the above are true, then the record must be fetched from the DWN.
+      this.#readableStream = this.#dwn.records.read(this.#target, { author: this.#author, message: { recordId: this.#recordId } })
+        .then((response) => response.record )
+        .then((record) => { return record.data; });
```

Comment:
why not just `(record) => record.data`?

Reply:
Sure, why not.  Addressed in 7a40a6622e2eb9fc885d89b83934f373ded6689f